pkg/stanza/adapter/receiver.go

```diff
@@ -132,15 +140,84 @@ func (r *receiver) consumerLoop(ctx context.Context) {
 				continue
 			}
 			obsrecvCtx := r.obsrecv.StartLogsOp(ctx)
-			cErr := r.consumer.ConsumeLogs(ctx, pLogs)
-			if cErr != nil {
-				r.logger.Error("ConsumeLogs() failed", zap.Error(cErr))
-			}
+			cErr := r.consumeLogs(ctx, pLogs)
 			r.obsrecv.EndLogsOp(obsrecvCtx, "stanza", pLogs.LogRecordCount(), cErr)
 		}
 	}
 }
 
+func (r *receiver) consumeLogs(ctx context.Context, logs plog.Logs) error {
```

Comment:
Is there anything about this functionality that makes it naturally unique to the stanza package? It seems quite generic to me. Should all components exhibit this behavior the same way?

Reply:
Yes, I think we can expose it in https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/coreinternal for now. Later we'll see if we can consolidate it with the retry functionality in core somewhere outside of the exporterhelper.

Reply:
Should we hold this PR until it's added in `coreinternal` then?

Reply:
I moved the logic to `coreinternal` in this PR